### DISCUSSION TITLE
Adjusting start times of all subtitles without affecting end times?

### DISCUSSION THREAD

**Discussion by fusefib** [SubtitleEdit/subtitleedit#7305] [Q&A] [answered]
Is there any simple method or feature to adjust the `Start time` of all subtitles to begin 200 milliseconds earlier, unless it results in overlap with the previous line's `End time`?

This operation would resemble the option `Adjust durations` -> `Adjust via` `Seconds` -> `Add seconds` `0.200`, but here it would be with the specific goal of extending the `Start time` in a backward direction rather than the `End time` in a forward direction.

Thanks for any help!

**Comment by Purfview** [comment]
Add -200ms global delay, then add +200ms in Tools>Adjust duration.

**Reply by fusefib** [reply]
Thanks for answering (and for whisper-faster!). Your operation's effect wouldn't be equivalent, as the intent is 
- keeping the end times intact, and
- not eating into the previous line's (preexisting) end timing when extending backwards. 

The first step with the global adjustment ignores this because there isn't a "wall" to hit. I'll show an example:

Original:
```
205
00:08:09,080 --> 00:08:11,066
{\an1}OR A COUPLE OF CARS. / 1.986, 0.024 to next

206
00:08:11,090 --> 00:08:12,610
{\an1}THEY ALSO SAID / 1.520, 0.480 to next
THAT EARLY THIS MORNING,

207
00:08:13,090 --> 00:08:14,250
{\an1}THEY HAD A REPORT / 1.160, 0.380 to next
ABOUT A HEAP

208
00:08:14,590 --> 00:08:16,066
{\an1}THAT FIT THE DESCRIPTION / 1.476, [0.024 to next]
OF PAUL'S
```
Desired effect with manual changes:
```
205
00:08:09,080 --> 00:08:11,066
{\an1}OR A COUPLE OF CARS. / 1.986, 0.024 to next

206
00:08:11,090 --> 00:08:12,610
{\an1}THEY ALSO SAID / 1.520, 0.080 to next
THAT EARLY THIS MORNING,

207
00:08:12,690 --> 00:08:14,250
{\an1}THEY HAD A REPORT / 1.560, 0.024 to next
ABOUT A HEAP

208
00:08:14,274 --> 00:08:16,066
{\an1}THAT FIT THE DESCRIPTION / 1.792, [0.024 to next]
OF PAUL'S
```
Your operation:
```
205
00:08:08,680 --> 00:08:10,666
{\an1}OR A COUPLE OF CARS. / 1.986, 0.024 to next

206
00:08:10,690 --> 00:08:12,610
{\an1}THEY ALSO SAID / 1.920, 0.080 to next
THAT EARLY THIS MORNING,

207
00:08:12,690 --> 00:08:14,166
{\an1}THEY HAD A REPORT / 1.476, 0.024 to next
ABOUT A HEAP

208
00:08:14,190 --> 00:08:15,666
{\an1}THAT FIT THE DESCRIPTION / 1.476, [0.024 to next]
OF PAUL'S
```

**Comment by Purfview** [comment, marked as answer]
> Your operation's effect wouldn't be equivalent

Yes, exact function like you described doesn't exist in SE, I just suggested something similar that maybe could fit your needs.

Anyway, your "Your operation" example doesn't look like something I described in my post.

**Reply by fusefib** [reply]
Thanks, I'll mark that as the answer - one can't easily do that for now.

Apologies, I used 400ms in the example and overlooked specifying that.

**Comment by oep42** [comment]
[Here](https://github.com/SubtitleEdit/subtitleedit/issues/6951) I suggested to add an option to "Adjust durations" to make each subtitle start earlier. I suggested exactly the same as what has been asked here (although it was in the context of "Batch Convert").

I hope that Nikolaj will consider to add such an option to "Adjust durations".

**Reply by fusefib** [reply]
Yes, it would be handy, though I'd rather not have it (only) in Batch Convert. On my end, I'll probably address it with a python script once I have some time to devote to the matter. You can probably even ask ChatGPT for one as it's not fundamentally complicated.
